presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java

```diff
@@ -171,37 +146,20 @@ public Void visitExchange(ExchangeNode node, Map<PlanNode, PlanNodeCost> context
                 }
             }
 
-            PlanNodeCost exchangeCost = PlanNodeCost.builder()
+            return PlanNodeCost.builder()
                     .setOutputRowCount(exchangeOutputRowCount)
                     .build();
-            context.put(node, exchangeCost);
-            return null;
         }
 
         @Override
-        public Void visitTableScan(TableScanNode node, Map<PlanNode, PlanNodeCost> context)
-        {
-            visitTableScan(node, context, BooleanLiteral.TRUE_LITERAL);
-            return null;
-        }
-
-        /**
-         * Return true if whole predicate has been enforced
-         */
-        private boolean visitTableScan(TableScanNode node, Map<PlanNode, PlanNodeCost> context, Expression predicate)
+        public PlanNodeCost visitTableScan(TableScanNode node, List<PlanNodeCost> context)
         {
-            boolean fullyEnforced = false;
-            // todo add mechanism to properly set fullEnforced flag.
-            // this requires extending metadata.getTableStatistic with return information on which part of passed constraind
-            // is enforced by table scan.
-            Constraint<ColumnHandle> constraint = getConstraint(node, predicate);
+            Constraint<ColumnHandle> constraint = getConstraint(node, BooleanLiteral.TRUE_LITERAL);
```

Comment:
Add todo for handling filter/scan scenario.
Probably we want some solution as part of this PR.